src/builtin_set_color.cpp

```diff
@@ -179,7 +179,9 @@ int builtin_set_color(parser_t &parser, io_streams_t &streams, wchar_t **argv) {
             write_color(rgb_color_t::black(), true /* is_fg */);
             writembs(tparm(exit_attribute_mode));
         } else {
-            write_color(fg, true /* is_fg */);
+            if (!write_color(fg, true /* is_fg */)) {
```

Comment:
This appears non-obvious and in need of a comment:

"If we can't write any color, write a reset sequence so cartesian products don't nix something like `echo (set_color foo)bar`. This _should_ work in monochrome terminals."